src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all books from DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books "
+                + "SET title = ?, price = ?"
+                + "WHERE id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update book with ID: "
+                    + book.getId() + " in DB", e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int rowsAffected = preparedStatement.executeUpdate();
+            return rowsAffected > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete book with ID: " + id + " from DB", e);
+        }
+    }
+
+    @Override
+    public boolean clearTable() {
+        String query = "TRUNCATE TABLE books;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            int rowsAffected = preparedStatement.executeUpdate();
+            return rowsAffected > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot truncate (clear) table in DB", e);
+        }
+    }
+
+    private Book getBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Long bookId = resultSet.getLong("id");
+        String title = resultSet.getString("title");
+        BigDecimal price = resultSet.getBigDecimal("price");
+
+        Book book = new Book();
+        book.setId(bookId);
+        book.setTitle(title);
+        book.setPrice(price);
+        return book;
```

Comment:
Avoid using both setters and constructor to set the properties of the object. Since you are using setters here, you should remove the constructor that sets these properties if it exists in the Book class, or use the constructor instead of the setters if that's more appropriate.